clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -601,6 +601,40 @@ public void closeShouldBeIdempotent() {
         producer.close();
     }
 
+    @Test
+    public void closeWithNegativeTimestampShouldThrow() {
+        Properties producerProps = new Properties();
+        producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9000");
+        try (Producer producer = new KafkaProducer<>(producerProps, new ByteArraySerializer(), new ByteArraySerializer())) {
+            assertThrows(IllegalArgumentException.class, () -> producer.close(Duration.ofMillis(-100)));
+        }
+    }
+
+    @Test
+    public void testFlush() {
```

Comment:
nit: make the test name more descriptive, like `testFlushCompleteSendOfInflightRecords`